### DISCUSSION TITLE
Plugin for SPDX / CycloneDX etc.

### DISCUSSION THREAD

**Discussion by hans-lammda** [Nordix/bomres#75] [Ideas]
Currently the focus on BOM-resolver is 

- Build everything from source code ( including tools for building the product ) 
- Build in a air-gapped environment 

Currently the format is JSON, but we should support SPDX or Cyclone/DX or both ?

One idea is to implement entry and exit classes with plugins. The current JSON model is very simple with just load/dump of the 
internal representation, additional plugins could then be added based on the design patterns listed below. 

- https://gist.github.com/dorneanu/cce1cd6711969d581873a88e0257e312
- https://mwax911.medium.com/building-a-plugin-architecture-with-python-7b4ab39ad4fc
- https://alysivji.github.io/simple-plugin-system.html

**Comment by stevespringett** [comment]
+1 for CycloneDX as it works with all our existing security tooling.
